spyder_kernels/comms/kernelcomm.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+In addition to the remote_call mechanism implemented in CommBase:
+ - Send a message to a debugging kernel
+"""
+import logging
+
+from qtpy.QtCore import QEventLoop, QObject, QTimer, Signal
+
+from spyder_kernels.comms.commbase import CommBase
+from spyder_kernels.py3compat import TimeoutError
+
+logger = logging.getLogger(__name__)
+
+
+class KernelComm(CommBase, QObject):
+    """
+    Class with the necessary attributes and methods to handle
+    communications with a console.
+    """
+
+    sig_got_reply = Signal(str)
+
+    def __init__(self):
+        super(KernelComm, self).__init__()
+
+        self.kernel_client = None
+        self._debugging = False
+        self._wait_list = {}
+
+        self.register_call_handler(
+            'set_debug', self._handle_debug_state)
```

Comment:
This should be `'set_debug_state'` because the meaning of `'set_debug'` is not so clear.